plugin-api/src/main/java/org/sonar/api/rules/RuleCharacteristic.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.rules;
+
+import org.sonar.api.Beta;
+
+/**
+ * Enum that represents Characteristic according to the new Clean Code Taxonomy. Should be used instead of {@link RuleType} as it will
+ * become mandatory in the future to provide it.
+ *
+ * @since 9.16
+ */
+@Beta
+public enum RuleCharacteristic {
+
+  // Fit for development
+  CLEAR, CONSISTENT, STRUCTURED, TESTED,
+  // Fit for production
```

Comment:
What about replacing the comments with types ? What I mean is introduce another enum that would contain DEVELOPMENT and PRODUCTION, and the characteristics would refer to those. It doesn't cost a lot, specifies the relationship between characteristics, and later if products want to filter rules/issues based on characteristic "kind", they could. Not sure how to name this concept though.

Reply:
If I understood correctly, the idea would be for this enum to have a field named "category"?

Reply:
Yes

Reply:
Hello, if I may, I would have one comment here.
In SC we are currently in an implementation sprint to display the naive mapping of rule characteristics. On our standup we were discussing exactly this - if the high level category of a characteristic should be defined already on the plugin API level, or if the relationship should be "hard coded" in the frontend code. I think it's a good idea to introduce the new enum type `Category` and  have it as a field of `RuleCharacteristic`. Thanks to that we will have access to this information already in the backend and can return it in the response, so no hard code on frontend. This information might be handy in some future use cases as well.

Reply:
Fixed